pkg/bid/bid.go

```diff
@@ -0,0 +1,113 @@
+package bid
+
+import (
+	"fmt"
+	"strings"
+
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+const (
+	BidPrefix            = "bid"
+	ResourceBidPrefix    = "r"
+	EntitlementBidPrefix = "e"
+	GrantBidPrefix       = "g"
+)
+
+type BID interface {
+	proto.Message
+	// TODO: be smarter about this interface.
+	GetAnnotations() []*anypb.Any
+}
+
+func MakeBid(b BID) (string, error) {
```

Comment:
style nit: can just be called `Make` or `New()`?  bid is duplicative of the package

Reply:
Looking this is really like an Encoder/Marhsler?

so kinda like `big.MarshalText`?

Reply:
Any recommendations for a shorter name? I called the other one bid.Parse(). Maybe bid.String()?